regression-test/suites/nereids_rules_p0/mv/increment_create/left_join_range_date_increment_create.groovy

```diff
@@ -0,0 +1,518 @@
+        group by l_shipdate, l_orderkey, o_orderdate, o_orderkey  """
+
+    def mv_sql_3 = """select l_shipdate, l_orderkey, o_orderdate, o_orderkey  
+        from lineitem_left_2
+        left join orders_left_2
+        on lineitem_left_2.l_shipdate = orders_left_2.o_orderdate
+        """
+
+    // no window func + on not partition col
+    def mv_sql_4 = """select l_shipdate, l_orderkey, o_orderdate, o_orderkey    
+        from lineitem_left_2 
+        left join orders_left_2 
+        on lineitem_left_2.l_orderkey = orders_left_2.o_orderkey 
+        group by l_shipdate, l_orderkey, o_orderdate, o_orderkey  """
+
+    def mv_sql_6 = """select l_shipdate, l_orderkey, o_orderdate, o_orderkey  
+        from lineitem_left_2
+        left join orders_left_2
+        on lineitem_left_2.l_orderkey = orders_left_2.o_orderkey
+        """
+
+    // window func
+    def mv_sql_7 = """select l_shipdate, l_orderkey, O_ORDERDATE, 
+        count(orders_left_2.O_ORDERDATE) over (partition by lineitem_left_2.L_SHIPDATE order by lineitem_left_2.L_ORDERKEY) as window_count 
+        from lineitem_left_2 
+        left join orders_left_2 
+        on lineitem_left_2.l_orderkey = orders_left_2.o_orderkey 
+        group by l_shipdate, l_orderkey, O_ORDERDATE"""
+
+    def mv_sql_8 = """select l_shipdate, l_orderkey, O_ORDERDATE,
+        count(orders_left_2.O_ORDERDATE) over (partition by lineitem_left_2.l_orderkey order by lineitem_left_2.l_orderkey) as window_count
+        from lineitem_left_2
+        left join orders_left_2
+        on lineitem_left_2.l_orderkey = orders_left_2.o_orderkey
+        group by l_shipdate, l_orderkey, O_ORDERDATE"""
+
+    def mv_sql_9 = """select l_shipdate, l_orderkey, O_ORDERDATE, 
+        count(orders_left_2.O_ORDERDATE) over (order by lineitem_left_2.l_orderkey) as window_count 
+        from lineitem_left_2 
+        left join orders_left_2 
+        on lineitem_left_2.l_orderkey = orders_left_2.o_orderkey 
+        group by l_shipdate, l_orderkey, O_ORDERDATE"""
+
+    def compare_res = { def stmt ->
+        def origin_res = sql stmt
+        logger.info("origin_res: " + origin_res)
+        def mv_origin_res = sql """select * from ${mv_name} order by 1,2,3,4"""
+        logger.info("mv_origin_res: " + mv_origin_res)
+        assertTrue((mv_origin_res == [] && origin_res == []) || (mv_origin_res.size() == origin_res.size()))
+        for (int row = 0; row < mv_origin_res.size(); row++) {
+            assertTrue(mv_origin_res[row].size() == origin_res[row].size())
+            for (int col = 0; col < mv_origin_res[row].size(); col++) {
+                assertTrue(mv_origin_res[row][col] == origin_res[row][col])
+            }
+        }
+    }
+
+    def sql_all_list = [mv_sql_1, mv_sql_3, mv_sql_4, mv_sql_6, mv_sql_7, mv_sql_8, mv_sql_9]
+    def sql_increment_list = [mv_sql_1, mv_sql_3, mv_sql_4, mv_sql_6, mv_sql_7]
+    def sql_complete_list = []
+
+    // create mv with partition col
```

Comment:
comment should be `base on left table with part col` would be better

Reply:
Modified comment